File: src/ldk_athlete_ai_coach/api/v1/schemas/metrics.py
from datetime import date

from pydantic import BaseModel, ConfigDict

from ldk_athlete_ai_coach.domain.models.training_metrics import TrainingMetrics


class MetricAdherenceResponse(BaseModel):
    """Response schema for adherence to a specific training metric."""

    model_config = ConfigDict(from_attributes=True)

    metric_name: str
    adherence_percentage: float | None


class TrainingMetricsResponse(BaseModel):
    """Response schema for training metrics of a given timeframe."""

    model_config = ConfigDict(from_attributes=True)

    timeframe_start: date | None
    timeframe_end: date | None
    training_metrics: TrainingMetrics

File: frontend/types/api/metrics.ts
import type { ApiDateString, WorkoutStatus } from "./common";

export interface MetricAdherence {
  metric_name: string;
  adherence_percentage: number | null;
}

export interface TrainingMetrics {
  planned_training_load: number;
  actual_training_load: number;
  metric_adherence: MetricAdherence[];
  included_statuses: WorkoutStatus[];
}

export interface MetricAdherenceResponse {
  metric_name: string;
  adherence_percentage: number | null;
}

export interface TrainingMetricsResponse {
  timeframe_start: ApiDateString | null;
  timeframe_end: ApiDateString | null;
  training_metrics: TrainingMetrics;
}
